### DISCUSSION TITLE
Report switch doesn't show the details of EC2 filter results when filtering by tags

### DISCUSSION THREAD

**Discussion by tarunaroraonline** [cloud-custodian/cloud-custodian#7364] [General]
# EC2 shutdown using Tags
---
I have associated a tag called maid_offhours to an EC2 instance, the value of the tag is `off=[(M-F,18),(U,17)];on=[(M-F,6),(U,10)];tz=bst`. In order to test this, I have created a policy, see the policy YAML below... 

> I have taken out the action block as I don't want to shut it down before I can test that the filter is working correctly. 

```yaml
policies:
  - name: stop-after-hours
    resource: ec2
    filters:
      - type: offhour
        tag: maid_offhours
        default_tz: bst
        offhour: 17
        opt-out: false
``` 
I should be able to use the `report` command to see the list of EC2 instances that would be covered by this policy? 

`custodian report -s ./infra.core/scripts/out/ ./infra.core/scripts/off-hours.yml --format grid --profile devtest`

instead i get this back (no results)... 

+-----------------+--------------+------------+----------------+--------------+---------+--------------------+
| CustodianDate   | InstanceId   | tag:Name   | InstanceType   | LaunchTime   | VpcId   | PrivateIpAddress   |
+=================+==============+============+================+==============+=========+====================+
+-----------------+--------------+------------+----------------+--------------+---------+--------------------+

If I run the command with dryrun, i can see that it manages to filter down to 1 instance...

`custodian run --dryrun -s ./infra.core/scripts/out/ ./infra.core/scripts/off-hours.yml --profile devtest`
```
2020-05-04 22:32:54,128: custodian.filters:INFO InstanceOffHour 1 of 1 instances
2020-05-04 22:32:54,129: custodian.policy:INFO policy:stop-after-hours resource:ec2 region:eu-west-1 count:0 time:0.53
```

If I run the command without dryrun, I understand this is fine as there is no action in my policy

`custodian run -s ./infra.core/scripts/out/ ./infra.core/scripts/off-hours.yml --profile devtest`
```
2020-05-04 22:33:08,028: custodian.filters:INFO InstanceOffHour 1 of 1 instances
2020-05-04 22:33:08,028: custodian.policy:INFO policy:stop-after-hours resource:ec2 region:eu-west-1 count:0 time:0.00
```

# Question
Why does the report filter not show the details of the EC2 instance that is being filtered down by the policy?

**Comment by kapilt** [comment]
the offhour filter only matches at the exact time of the filter.  report formats policy output from a previous run. ie you if you hadn't run the poicy before.

also perhaps its helpful to run with --verbose when running. also keep in mind there's a default cache period of 15m if your manipulating the environment when running on the cli. 

its also useful to look at the docs here, for things that can help (like skew parameter for offhours).
